git-remote-hg

```diff
@@ -265,10 +266,10 @@ class Parser:
         return (user, int(date), -tz)
 
 def fix_file_path(path):
-    path = os.path.normpath(path)
+    path = posixpath.normpath(path)
     if not os.path.isabs(path):
```

Comment:
Sorry for my English, it's not my native language. I was in a hurry to answer you and wrote nonsense, but now I will write thoughtfully and use the translator (dictionary) for compound words. I hope now you will understand my words. ;-) 
Unfortunately, razaraz is the author of the commit but do not pull request and unlikely he will be read here. Because this pull request started me, I need to justify it. 
These two line changes enough to make your program run on Windows better. With this commit operation start working "clone", "pull" and "checkout" is correct. With this correction operation start working three main commands correctly. Maybe there are other problems, but it did not work on Windows if there is a directory in the repository.
I go with the practical side of the issue, correction harmless for Linux and MAC OS, and solves part of the serious problems on Windows. Let other authors fix other lines, now it is premature because it will correct other errors, not #4.
Of course, you want to in-depth study of the problem. But one mistake can be corrected now. However, this fix will help many people now, without spoiling anything on Linux and MAC OS.
Maybe the best fix will never be done, if not find someone with free time.
As you can see in the above error messages, there is no more error with text "Filenames in Git repo have \ as file separator on Windows" neither I nor nathanielcook in operation "checkout". Other errors, if they were, should be corrected in other commits.
If you wish (and have the time), I can prepare a public repository with tests that show errors with other operations (push). Unfortunately, I do not have much experience of development in Python, to improve such a complex program, but I can give the log output to the console on Windows, and attach the file to the folder ".git" for different experiments, when an error occurs.
As I understand the idea of open sourse, will make the most correct merge with the original commit of razaraz, and then add our commits, which will make the program even more stable and reliable.
Thank you for reading this. I hope I was convincing and understandable. Sorry if the the text view larger. Thanks for develop this program for the world.